libs/wingc/src/type_check.rs

```diff
@@ -2045,6 +2062,55 @@ impl<'a> TypeChecker<'a> {
 				self.inner_scopes.push(statements);
 			}
 			StmtKind::Break | StmtKind::Continue => {}
+			StmtKind::IfLet {
+				condition,
+				statements,
+				var_name,
+				else_statements,
+			} => {
+				let cond_type = self.type_check_exp(condition, env);
+				self.validate_type_is_optional(cond_type, condition);
+
+				// Technically we only allow if let statements to be used with optionals
+				// and above validate_type_is_optional method will attach a diagnostic error if it is not.
+				// However to avoid panics this will allow code to continue if the type is not an optional
```

Comment:
Actually we're not avoiding panics, we could just return here after creating the diagnostic, but we continue anyway for the sake of verbose diagnostics (see #884).
```suggestion
				// However for the sake of verbose diagnostics we'll allow the code to continue if the type is not an optional
```